src/Uno.UI/UI/Xaml/Documents/TextElement.cs

```diff
@@ -345,7 +345,7 @@ void SetDefaultForeground(DependencyProperty foregroundProperty)
 			{
 				// Hyperlink doesn't appear to inherit foreground from the parent.
 				// So, we set this with ImplicitStyle precedence which is a higher precedence than Inheritance.
-				this.SetValue(foregroundProperty, DefaultTextForegroundBrush, DependencyPropertyValuePrecedences.ImplicitStyle);
+				this.SetValue(foregroundProperty, DefaultTextForegroundBrush, DependencyPropertyValuePrecedences.DefaultStyle);
```

Comment:
The right way to do that is by using `this.ForegroundProperty.OverrideMetadata()`.